azure-ts-appservice/index.ts

```diff
@@ -0,0 +1,104 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as azure from "@pulumi/azure";
+import { signedBlobReadUrl } from "./sas";
+
+const name = 'azpulumi';
+
+const resourceGroup = new azure.core.ResourceGroup(`${name}-rg`, {
+        location: "West US 2",
+    });
+
+const resourceGroupArgs = {
+    resourceGroupName: resourceGroup.name,
+    location: resourceGroup.location,
+};
+
+const storageAccount = new azure.storage.Account(`${name}sa`, {
+    ...resourceGroupArgs,
+
+    accountKind: "StorageV2",
+    accountTier: "Standard",
+    accountReplicationType: "LRS",
+});
+
+
+const appServicePlan = new azure.appservice.Plan(`${name}-asp`, {
+    ...resourceGroupArgs,
+
+    kind: "App",
+
+    sku: {
+        tier: "Basic",
+        size: "B1",
+    },
+});
+
+const storageContainer = new azure.storage.Container(`${name}-c`, {
+    resourceGroupName: resourceGroup.name,
+    storageAccountName: storageAccount.name,
+    containerAccessType: "private",
+});
+
+const map: pulumi.asset.AssetMap = {};
+map["index.html"] = new pulumi.asset.StringAsset("<html><body><h1>Greetings from Azure App Service!</h1></body></html>");
+
+const blob = new azure.storage.ZipBlob(`${name}-b`, {
+    resourceGroupName: resourceGroup.name,
+    storageAccountName: storageAccount.name,
+    storageContainerName: storageContainer.name,
+    type: "block",
+
+    content: new pulumi.asset.AssetArchive(map)
+             // to pack a folder use new pulumi.asset.FileArchive("folder")
```

Comment:
Is it worth doing the folder mode by default here?  I assume that will be more "normal".

Reply:
In my real example I had an ASP.NET Core app there, but wasn't sure just making a folder with `index.html` would make sense

Reply:
Yeah - I think a folder with just an `index.html` makes sense - and is closer to what most folks (including yourself!) would want to do here.

This pattern of building up the archive in memory appears to be borrowed from the Functions example, but was only used there because that code actually is dynamically constructing the contents to include in the upload as part of the program.  In the case of App Service here, it would be very rare I expect to need/want to do that.  This will also remove some complexity here.